### PR TITLE
Upgrade pigeon

### LINKED ISSUE #209: Update Pigeon models to use non-nullable generics
The release of Pigeon 22.4.0 allows for non-nullable generics in collections (i.e. `List`, `Map`). We should upgrade Pigeon and refactor the models to take advantage of this as dealing with nullable generics is a bit of a pain in the Dart-side marshalling.

### PR DESCRIPTION
**Pull request by TomRHandcock**
- Upgrades `pigeon` dependency in `mockzilla_android` + `mockzilla_ios`.
- Fixes an issue where headers were not correctly marshalled across pigeon bridge.

Merging this PR will resolve #209

### CODE REVIEW

File: FlutterMockzilla/mockzilla_android/example/pubspec.lock [hunk 1]
```diff
@@ -153,14 +153,14 @@ packages:
...
-    version: "0.1.2"
+    version: "1.0.0-dev.1"
   mockzilla_platform_interface:
     dependency: "direct overridden"
     description:
       path: "../../mockzilla_platform_interface"
       relative: true
     source: path
-    version: "0.2.0"
+    version: "1.0.0-dev.0"
```
Comment (SamC-Apadmi) [inline FlutterMockzilla/mockzilla_android/example/pubspec.lock:163]: This doesn't need to be dev.1 does it?

File: FlutterMockzilla/mockzilla_android/lib/src/model/mockzilla_error.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  get details => key;
+
+  @override
+  String? get message =>
+      "Mockzilla tried to find an endpoint with key $key but "
+          "it doesn't exist.";
+
+  @override
+  String? get stacktrace => trace.toString();
+}
```
Comment (SamC-Apadmi) [inline FlutterMockzilla/mockzilla_android/lib/src/model/mockzilla_error.dart:24]: 👍

**Review by SamC-Apadmi — APPROVED** [review]

**Review by chedabob — APPROVED** [review]

File: FlutterMockzilla/mockzilla_android/example/pubspec.lock (hunk 1 above)
Comment (TomRHandcock) [reply FlutterMockzilla/mockzilla_android/example/pubspec.lock:163]: This is a dependency on the platform interface package which hasn't been subject to any changes. Further, this is just the example app which is only really present to provide the Gradle tasks necessary to run unit tests; the _actual_ examples is in the main package.

Comment (SamC-Apadmi) [reply FlutterMockzilla/mockzilla_android/example/pubspec.lock:163]: Gotcha!

**Pull request merged**